### PR TITLE
chore: add CONTRIBUTING.md draft

### PR DESCRIPTION
**Pull request by baggiponte**
Adding a draft for CONTRIBUTING.md. Would like to ask for a review to @ngriffiths13, @FBruzzesi, @miroslaavi who is working on a PR and installed the project locally, and also @ rodrigogiraoserrao because he says they are important (🤪).

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+* **Start your pull request title with a [conventional commit tag](https://www.conventionalcommits.org/en/v1.0.0/)**. This helps us add your contribution to the right section of the changelog. We use the [Angular](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type) convention.
+* Use a descriptive title starting with an uppercase letter. This text will end up in the changelog.
+* In the pull request description, [link](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) to the issue you were working on.
+* Add any relevant information to the description that you think may help the maintainers review your code.
+* Make sure your branch is [rebased](https://docs.github.com/en/get-started/using-git/about-git-rebase) against the latest version of the main branch.
+* Make sure all GitHub Actions checks pass.
+* After you have opened your pull request, a maintainer will review it and possibly leave some comments. Once all issues are resolved, the maintainer will merge your pull request, and your work will be part of the next Polars release!
```
Comment (miroslaavi) [inline CONTRIBUTING.md:104]:
```suggestion
* After you have opened your pull request, a maintainer will review it and possibly leave some comments. Once all issues are resolved, the maintainer will merge your pull request, and your work will be part of the next Functime release!
```

**Review by miroslaavi — APPROVED** [review]
I guess I was an inspiration with my mess PR to add the contributing guide, hahah! :D but it looks great and very useful!

Just thoughts, maybe for new user it could be helpful having an example how the conventional commit would look like. Also, not sure if branch naming matters?

**Comment by baggiponte** [comment]
Thanks for the comments! I was actually curious how you set the project up.

I'm a bit obsessed with the conventional commits but I'm sure we also use it for our changelog, so it has a purpose. Will add that part 😊

Speaking of branches: I don't think it really matters as long as it is understandable. Will mention that. (Also I did not forget about the PR but this is something that I postponed a bit too much.)

**Comment by miroslaavi** [comment]
Cool, that's clear, thanks!

Hmm, I set the project up by first downloading Rust (as that is something I have no prior experience of, but have started thinking about learning it), and then I think I just ran the Makefile "edit" if I recall right.. Rye also something new for me that need to look up :)

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+```bash
+gh repo fork functime-org/functime
+```
+
+The CLI will prompt you to clone the fork locally.
+
+2. **Clone the repository locally**.
+
+```bash
```
Comment (FBruzzesi) [inline CONTRIBUTING.md:42]: More ideas, make one expanded and the other two collapsed html _details_

**Review by FBruzzesi — COMMENTED** [review]
Hey @baggiponte, thanks for including me in the review.
I left a few nitpicks and some food for thoughts 😁

**Pull request merged**